---
title: "ChIP-seq Peak Processing and Annotation"
output: html_document
date: "2025-02-07"
---


library(GenomicRanges)
library(devtools)
library(bedr)


```

# Read peak files into GRanges objects
d0_WT_BMI1   <- bed_to_granges("WT_d0_BMI1_peaks.narrowPeak")
d0_WT_RING1B <- bed_to_granges("WT_d0_RING1B_peaks.narrowPeak")
d0_WT_SCMH1  <- bed_to_granges("WT_d0_SCMH1_peaks.narrowPeak")

d0_KO_BMI1   <- bed_to_granges("KO_d0_BMI1_peaks.narrowPeak")
d0_KO_RING1B <- bed_to_granges("KO_d0_RING1B_peaks.narrowPeak")
d0_KO_SCMH1  <- bed_to_granges("KO_d0_SCMH1_peaks.narrowPeak")

d4_WT_BMI1   <- bed_to_granges("WT_d4_BMI1_peaks.narrowPeak")
d4_WT_RING1B <- bed_to_granges("WT_d4_RING1B_peaks.narrowPeak")
d4_WT_SCMH1  <- bed_to_granges("WT_d4_SCMH1_peaks.narrowPeak")
d4_WT_IgG    <- bed_to_granges("WT_d4_IgG_control.narrowPeak")

d4_KO_BMI1   <- bed_to_granges("KO_d4_BMI1_peaks.narrowPeak")
d4_KO_RING1B <- bed_to_granges("KO_d4_RING1B_peaks.narrowPeak")
d4_KO_SCMH1  <- bed_to_granges("KO_d4_SCMH1_peaks.narrowPeak")
d4_KO_IgG    <- bed_to_granges("KO_d4_IgG_control.narrowPeak")

#Remove IgG background peaks
d4.S1KO.BMI1.minuscontrol <- subsetByOverlaps(d4.S1KO.BMI1, d4.S1KO.IgG, invert = T)
d4.WT.BMI1.minuscontrol <- subsetByOverlaps(d4.WT.BMI1, d4.WT.IgG, invert = T)
d4.S1KO.RING1B.minuscontrol <- subsetByOverlaps(d4.S1KO.RING1B,d4.S1KO.IgG, invert = T)
d4.WT.RING1B.minuscontrol <- subsetByOverlaps(d4.WT.RING1B, d4.WT.IgG, invert = T)
d4.S1KO.SCMH1.minuscontrol <- subsetByOverlaps(d4.S1KO.SCMH1,d4.S1KO.IgG, invert = T)
d4.WT.SCMH1.minuscontrol <- subsetByOverlaps(d4.WT.SCMH1, d4.WT.IgG, invert = T)

#Identify WT specific peaks
d0.WT.SCMH1.peaks <- subsetByOverlaps(subsetByOverlaps(d0.WT.SCMH1, d0.S1KO.SCMH1, invert = T), d4.S1KO.SCMH1.minuscontrol,invert=T)
d4.WT.SCMH1.peaks <- subsetByOverlaps(subsetByOverlaps(d4.WT.SCMH1.minuscontrol, d4.S1KO.SCMH1.minuscontrol, invert = T), d0.S1KO.SCMH1,invert=T)

#Identify shared binding targets (SCMH1 and RING1B/BMI1)
shared_targets_BMI1_SCMH1 <- subsetByOverlaps(d4.WT.SCMH1.peaks, d4.WT.BMI1.minuscontrol)
shared_targets_RING1B_SCMH1 <- subsetByOverlaps(d4.WT.SCMH1.peaks, d4.WT.RING1B.minuscontrol) 
shared_BMI_RING1B_SCMH1 <- subsetByOverlaps(shared_targets_BMI1_SCMH1, shared_targets_RING1B_SCMH1) 

annotation<-import.gff("GRCm39.refseq_select.gtf")

#Map peaks to nearest promoter

annotation.transcript=annotation[annotation$type=="transcript"]
annotation.plus=annotation.transcript[strand(annotation.transcript)=="+"]
annotation.minus=annotation.transcript[strand(annotation.transcript)=="-"]

window=2500
promoter=GRanges(seqnames=c(seqnames(annotation.plus),seqnames(annotation.minus)),
                 ranges=IRanges(start=c(start(annotation.plus)-window,end(annotation.minus)-window),
                                end=c(start(annotation.plus)+window,end(annotation.minus)+window)),
                 strand=c(rep("+",length(annotation.plus)),rep("-",length(annotation.minus))),
                 gene=c(annotation.plus$gene,annotation.minus$gene))


near_promoter <- distanceToNearest(shared_BMI_RING1B_SCMH1,promoter)
distances <- mcols(near_promoter)$distance

#filter out genes over 50 kb away

near_promoter_filtered = near_promoter[which(distances < 50000)]

#Final list of targets
chipseq_genes <- promoter$gene[subjectHits(near_promoter_filtered)]
near_promoter_ring1b <- distanceToNearest(d4.WT.RING1B.minuscontrol,promoter)
distances_ring1b <- mcols(near_promoter_ring1b)$distance
near_promoter_ring1b_filtered <- near_promoter[which(distances_ring1b < 50000)]
ring1b_genes <- promoter$gene[subjectHits(near_promoter_ring1b_filtered)]

